### PR TITLE
Embed font files in Odoc 

### PR DESCRIPTION
**Pull request by Julow**
Closes https://github.com/ocaml/odoc/issues/694

Remove the remote fonts from the default theme and instead embed them in the theme.

This removes requests to Google's server that users might not consent to.

The font files are part of the theme, they are affected by `--without-theme` and are installed in `share/odoc/odoc-theme/default/`.
The lack of weight variants don't seem to be a problem.

Fira Mono and Fira Sans are from the latest release on https://github.com/mozilla/Fira.
Noticia Text is from the latest release on Google Fonts.

Notes:

- This should not be merge now as it would break the doc built with Dune.
  We should merge this after the release to leave time to update Dune to use `support-files-targets`.

- I removed `highlightjs` from the files installed in `prefix/share`.
  A "theme" should be just the files that are allowed to be tweaked: the CSS and text fonts.
  More consistent with `odoc support-files --without-theme`. If we wish to make highlightjs optional, we should add a flag for that.

- The CSS file is not in the same place within Odoc's source due to some refactoring. I hope no one was relying on that.

### CODE REVIEW

**Comment by dbuenzli** [comment]
> The lack of weight variants don't seem to be a problem.

Are you sure you are not falling back on browser font synthesis for bolds and italics (you can check by adding `:root { font-synthesis: none }` in the CSS) ? If that is the case, there's a high probability that type extremists will show up scandalized on the issue tracker (which is, as far as I'm concerned, not an entirely unfair behaviour).

**Comment by Julow** [comment]
You are right, font synthesis is used. There's a few `font-weight: 500` on headings, the TOC and keywords in the code. This rule is shadowed on headings but the difference on keywords and in the TOC is clearly visible. The font synthesis don't render the weight `500` differently from `400` on my monitor at 300% zoom on Firefox.
The previous font had a difference here. It seems that we need 3 more font files ?

- Fira Sans weight 500 (H1 in TOC and H0 and I guess it was intended on H1 in general)
- Fira Mono weight 500 (keywords)
- Noticia Text weight 700 (bold markup)

**Comment by Julow** [comment]
I moved the commits not related to the fonts into https://github.com/ocaml/odoc/pull/915 in case they can be merged sooner.

**Comment by jonludlam** [comment]
We were missing some fonts - notably the italic ones - so I've added a few more. The original css had:

```css
@import url('https://fonts.googleapis.com/css?family=Fira+Mono:400,500');
@import url('https://fonts.googleapis.com/css?family=Noticia+Text:400,400i,700');
@import url('https://fonts.googleapis.com/css?family=Fira+Sans:400,400i,500,500i,600,600i,700,700i');
```

so I've got all of these minus the 600-weight fonts. Looking back, these fonts were first introduced in https://github.com/ocaml/odoc/pull/208 - and that same commit removed all mention of 600 in the font weights, so I don't think we ever used them.

File: src/html_support_files/odoc.css [hunk 1]
```diff
@@ -4,9 +4,105 @@
    %%NAME%% %%VERSION%% */
 
 /* Fonts */
-@import url('https://fonts.googleapis.com/css?family=Fira+Mono:400,500');
-@import url('https://fonts.googleapis.com/css?family=Noticia+Text:400,400i,700');
-@import url('https://fonts.googleapis.com/css?family=Fira+Sans:400,400i,500,500i,600,600i,700,700i');
+/* noticia-text-regular - latin */
+@font-face {
+  font-display: swap; /* Check https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display for other options. */
```
Comment (Julow) [inline src/html_support_files/odoc.css:9]: I don't think we should configure this. The default is probably the best as we generate a static website with nothing dynamic.

**Review by Julow — COMMENTED** [review]
Happy with all the font files. Is this ready for merge ?

File: src/html_support_files/odoc.css (hunk 1 above)
Comment (jonludlam) [reply src/html_support_files/odoc.css:9]: Fair enough - I don't have a strong opinion and was just using the default from google-webfonts-helper. Using the default seems reasonable to me, so I'll push a fix. Thanks!

**Comment by jonludlam** [comment]
I think this is ready. In it goes - thanks, @Julow !

**Pull request merged**